src/annflux/training/annflux/clip_server.py

```diff
@@ -0,0 +1,201 @@
+            "version": "repo:model:8d3aca47",  # TODO
+            "tag": "StreetSurfaceVis",  # TODO
+        },
+        "media": [],
+        "region_groups": [],
+        "predictions": [],
+    }
+
+    # for each uploaded image perform classification
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        for uploaded_file in uploaded_files:
+            if uploaded_file.filename == "":
+                abort_json("received_no_files", "Did not receive any files", 400)
+
+            if not uploaded_file.filename.lower().endswith(EXTENSIONS):
+                # unsupported media type
+                abort_json("unsupported_media_type", "Unsupported media type", 415)
+
+            # - write file to temporary location
+            image_basename = os.path.basename(uploaded_file.filename)
+            image_id = os.path.splitext(image_basename)[0]
+            tmp_file = os.path.join(tmp_dir, image_basename)
+            with open(tmp_file, "wb") as f:
+                f.write(uploaded_file.read())
+
+            out_json["media"].append({"id": image_id, "filename": image_basename})
+            region_group_id = "region_0"
+            out_json["region_groups"].append(
+                {
+                    "id": region_group_id,
+                    "individual_id": region_group_id,
+                    "regions": ["(full)"],
+                }
+            )
+
+            for r, row in server.predict(tmp_file).iterrows():
+                out_json["predictions"].append(
+                    {
+                        "region_group_id": region_group_id,
+                        "classes": {
+                            "type": "multiclass",
+                            "items": [
+                                {
+                                    "name": row.predictions,
+                                    "probability": row.probability,
+                                }
+                            ],
+                        },
+                    }
+                )
+
+    return out_json
+
+
+def abort_json(error_code, error_message, http_status_code):
+    """
+    Aborts the response, returning a JSON response
+    :param http_status_code: the HTTP status code to return in the response
+    """
+    abort(standard_json_response(error_code, error_message, http_status_code))
+
+
+def standard_json_response(error_code, error_message, http_status_code):
+    if error_message is None:
+        error_message = " ".join(error_code.split("_")).capitalize()
+    logger.info("status: {}, error message: {}".format(http_status_code, error_message))
+    response = make_response(
+        flask.jsonify({"error": {"code": error_code, "message": error_message}}),
+        http_status_code,
+    )
+    return response
+
+
+def m(model_folder):
+    global server
+    server = ClipServer(model_folder)
+
+    app.run(
+        debug=True,  # str2bool(os.getenv("APP_DEBUG", False)),
+        host="0.0.0.0",
```

Comment:
Please use `127.0.0.1` (app will be accessible to process running on the same machine) - to reduce risk as `0.0.0.0` opens app externally.
```suggestion
        host="127.0.0.1",
```